File: backend/src/main/java/online/nonamelab/WorkSite_Work/exception/ApiError.java
package online.nonamelab.WorkSite_Work.exception;

import java.time.LocalDateTime;
import java.util.List;

public record ApiError(
        int status,
        String error,
        String message,
        String path,
        LocalDateTime timestamp,
        List<ApiFieldError> fieldErrors
) {
}

File: frontend/src/api/apiError.ts
import axios from "axios";

export type ApiFieldError = {
  field: string;
  message: string;
};

export type ApiErrorResponse = {
  status: number;
  error: string;
  message: string;
  path: string;
  timestamp: string;
  fieldErrors: ApiFieldError[];
};

export function getApiErrorMessage(error: unknown) {
  if (axios.isAxiosError<ApiErrorResponse>(error)) {
    return error.response?.data.message ?? "Request failed";
  }

  return "Unexpected error";
}

export function getApiFieldErrors(error: unknown) {
  if (axios.isAxiosError<ApiErrorResponse>(error)) {
    return error.response?.data.fieldErrors ?? [];
  }

  return [];
}
